### PR TITLE
filter logs only from local modules for prod build

### LINKED ISSUE #2040: Revisit logs in the worker
### Context

Things to check:
- log levels: `dev` vs `prod`
- logging of sensitive information: `dev` vs `prod`
- remove redundant logs: our worker log size grows rapidly

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by kziemianek**
Initializes `enclave-runtime` prod logger to allow log entries only from local modules.

close: #2040

### CODE REVIEW

File: tee-worker/litentry/macros/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			}
+	};
+	output.into()
+}
+
+fn read_module_names(path: &str, relative_to: &str, module_names: &mut Vec<String>) {
+	let current_path = relative_to.to_string() + "/" + path;
+	let cargo_file = current_path.to_string() + "/Cargo.toml";
+	let contents = fs::read_to_string(&cargo_file)
+		.unwrap_or_else(|_| panic!("Should have been able to read the file: {}", cargo_file));
```
Comment (Kailai-Wang) [inline tee-worker/litentry/macros/src/lib.rs:41]: Maybe `expect`?

**Review by Kailai-Wang — APPROVED** [review]
Looks good, thanks!

Have you tested that it actually worked? 🙌

File: tee-worker/litentry/macros/src/lib.rs (hunk 1 above)
Comment (kziemianek) [reply tee-worker/litentry/macros/src/lib.rs:41]: I had it before but clippy complained:

```
error: use of `expect` followed by a function call
  --> litentry/macros/src/lib.rs:41:4
   |
41 |         .expect(&format!("Should have been able to read the file: {}", cargo_file));
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|_| panic!("Should have been able to read the file: {}", cargo_file))`
```

Comment (Kailai-Wang) [reply tee-worker/litentry/macros/src/lib.rs:41]: I see, thanks! Learned something 👍

**Comment by kziemianek** [comment]
> Looks good, thanks!
> 
> Have you tested that it actually worked? 🙌

I checked it manually at runtime for non-prod build, it produced:
```
Initializing logger to filter only following local modules: ["enclave_runtime", "ita_oracle", "itc_rest_client", "itp_enclave_metrics", "ita_stf", "ita_sgx_runtime", "itp_sgx_runtime_primitives", "litentry_primitives", "itp_utils", "core_primitives", "pallet_identity_management_tee", "pallet_parentchain", "itp_hashing", "itp_node_api", "itp_api_client_extensions", "itp_api_client_types", "itp_types", "rococo_parachain_runtime", "pallet_asset_manager", "pallet_bridge", "pallet_bridge_transfer", "pallet_parachain_staking", "pallet_drop3", "pallet_evm_address", "pallet_extrinsic_filter", "pallet_group", "pallet_identity_management", "pallet_teerex", "sgx_verify", "teerex_primitives", "common_primitives", "test_utils", "pallet_sidechain", "sidechain_primitives", "pallet_teeracle", "teeracle_primitives", "pallet_vc_management", "runtime_common", "itp_node_api_factory", "itp_node_api_metadata", "itp_node_api_metadata_provider", "itp_sgx_externalities", "environmental", "itp_stf_interface", "itp_stf_primitives", "itp_storage", "lc_scheduled_enclave", "itp_component_container", "itp_settings", "itp_sgx_io", "lc_stf_task_sender", "itp_ocall_api", "itp_sgx_crypto", "itp_sgx_temp_dir", "sp_io", "itc_direct_rpc_server", "itc_tls_websocket_server", "itp_rpc", "itc_offchain_worker_executor", "itc_parentchain_light_client", "itc_parentchain_test", "itp_test", "itp_stf_state_handler", "itp_stf_state_observer", "itp_time_utils", "itp_teerex_storage", "itp_extrinsics_factory", "itp_nonce_cache", "itp_stf_executor", "itp_top_pool_author", "itp_top_pool", "its_primitives", "itc_parentchain", "itc_parentchain_block_import_dispatcher", "itc_parentchain_block_importer", "itc_parentchain_indirect_calls_executor", "itp_import_queue", "itp_attestation_handler", "itp_primitives_cache", "its_block_verification", "its_sidechain", "its_block_composer", "its_state", "its_consensus_aura", "its_consensus_common", "fork_tree", "its_consensus_slots", "its_validateer_fetch", "its_rpc_handler", "lc_data_providers", "lc_stf_task_receiver", "lc_assertion_build", "lc_credentials", "lc_identity_verification", "litentry_macros"]
```
In my opinion the list looks fine. 

I also considered putting this list in code, so it would be more explicit for us which modules will produce the logs, and intersecting it at runtime with the one produced from macro and eventually printing out the difference. If you think it might be worth to do so I can change the code.

**Comment by Kailai-Wang** [comment]
It seems it doesn't take features into account, and probably it can't do so anyway in the runtime.

We can leave it as it is for now

**Comment by kziemianek** [comment]
We can get a list of activated features from env variables at compilation time.

```
CARGO_FEATURE_<name> — For each activated feature of the package being built, this environment variable will be present where <name> is the name of the feature uppercased and having - translated to _.
```

I'm leaving it here as a tip for potential improvement.

**Pull request merged**